src/features/useCurrentPartyRoles.ts

```diff
@@ -1,16 +1,23 @@
 import { useQuery, type UseQueryOptions } from '@tanstack/react-query';
 
 import { useApplicationMetadata } from 'src/features/applicationMetadata/ApplicationMetadataProvider';
+import { useNavigationParam } from 'src/features/routing/AppRoutingContext';
 import { fetchRoles } from 'src/queries/queries';
 import { isAtLeastVersion } from 'src/utils/versionCompare';
 import type { ApplicationMetadata } from 'src/features/applicationMetadata/types';
 import type { Role } from 'src/types/shared';
 
 export type RoleResult = { data: Role[] | undefined; error: Error | null };
 
-export function useCurrentPartyRolesQueryDef(enabled: boolean): UseQueryOptions<Role[], Error> {
+export function useCurrentPartyRolesQueryDef(
+  supportsRolesAPI: boolean,
+  partyID?: string,
+  instanceGUID?: string,
+): UseQueryOptions<Role[], Error> {
+  const enabled = !!(supportsRolesAPI && partyID && instanceGUID);
+
   return {
-    queryKey: ['fetchCurrentPartyRoles'],
+    queryKey: [`fetchCurrentPartyRoles/${partyID}/${instanceGUID}`],
```

Comment:
Query keys support multiple values already, you can simply put them in the array.
```suggestion
    queryKey: ['fetchCurrentPartyRoles', partyID, instanceGUID],
```